### ISSUE TITLE
RuntimeError: Cannot run the event loop while another loop is running

### ISSUE THREAD

**Issue by mastanbol** [jreese/aiomultiprocess#4]
I am runnig your test case **test_pool_concurrency()** method which is there in **PerfTest.py**. But unfortunately I am getting a runtime error, PFB for the detailed error,
-------------------------------------------------Detailed Error-------------------------
**RuntimeError: Cannot run the event loop while another loop is running**
aio process 8186 failed
Traceback (most recent call last):
  File "/home/mastan/env2/lib/python3.6/site-packages/aiomultiprocess/core.py", line 103, in run_async
    self.aio_target(*self.aio_args, **self.aio_kwargs)
  File "/usr/local/lib/python3.6/asyncio/base_events.py", line 454, in run_until_complete
    self.run_forever()
  File "/usr/local/lib/python3.6/asyncio/base_events.py", line 411, in run_forever
    'Cannot run the event loop while another loop is running')
RuntimeError: Cannot run the event loop while another loop is running
Process ForkProcess-225:
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/local/lib/python3.6/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/home/mastan/env2/lib/python3.6/site-packages/aiomultiprocess/core.py", line 103, in run_async
    self.aio_target(*self.aio_args, **self.aio_kwargs)
  File "/usr/local/lib/python3.6/asyncio/base_events.py", line 454, in run_until_complete
    self.run_forever()
  File "/usr/local/lib/python3.6/asyncio/base_events.py", line 411, in run_forever
    'Cannot run the event loop while another loop is running')
RuntimeError: Cannot run the event loop while another loop is running
aio process 8187 failed 

**Steps followed:**
-> from perf import PerfTest
-> pt = PerfTest()
-> pt.test_pool_concurrency()

**Same thing is happening when i try to use aiomultiprocess in my present code.Could you please let me know if i am doing anything wrong,**

### Details
----------------
**My present scenario:**

I am using multiprocessing with pool.starmap() to process the records which i am reading from another resource. I am creating processes based on cpucount() of that machine ,for example 10 processes are there, then creating same no of db connections to give each record and db connection as tuple for a process to call the actual method.
```

from multiprocessing.dummy import Pool

def process_record(record, conn):
    # actual processing

def main():
	processes = cpu_count() * 5
    db_list = []
    for p in range(processes):
        db_conn_list.append(get_db_connection())
    print("Initializing the pool with " + str(processes) + " processes")
    pool = Pool(processes)
    try:
        response = self.client.get_records(ShardIterator=self.iterator)
        records = response.get('Records')
        arguments = zip(records[], db_conn_list)
        pool.starmap(process_record, arguments)
    except Exception as e:
        pool.close()
        pool.join()
        time.sleep(1)

if __name__ == '__main__':
    main()
```

--------------------------------------
**Using aioprocess in my current code:**
```

from aiomultiprocess import Pool

async def process_record(record, engine):
    processor = BitPostProcessor(conn_string, engine)
    await processor.insert_record(record)

async def main():
	processes = cpu_count() * 5
    db_list = []
    for p in range(processes):
        db_conn_list.append(get_db_connection())
    print("Initializing the pool with " + str(processes) + " processes")
    pool = Pool(processes)
    try:
        response = self.client.get_records(ShardIterator=self.iterator)
        records = response.get('Records')
        arguments = zip(records[], db_conn_list)
        await pool.starmap(process_record, arguments)
    except Exception as e:
        pool.close()
        pool.join()
        time.sleep(1)

```
if __name__ == '__main__':
    asyncio.get_event_loop().run_until_complete(main())

* OS: Ubuntu 16.04
* Python version: python 3.6.3
* aiomultiprocess version:0.4.0
* Can you repro on master? Yes, using test case
* Can you repro in a clean virtualenv? yes, i can

**Comment by jreese** [comment]
I'm not able to repoduce this error on either Mac or Ubuntu 18.04 when using Python 3.6.5.  Are you using any modules that change the way asyncio works, or using a third party asyncio event loop like uvloop?

**Comment by Gurubol** [comment]
@jreese , We are running exact same code above in a standalone file. Can you please suggest if we are missing something?

**Comment by jreese** [comment]
I've created a toy script following your example code, but without the database parts, and it works just fine.  See https://gist.github.com/jreese/73e05b7e407f42bb748b406f42e3abd6.

It sounds like your code, or one of your dependencies, is automatically creating an asyncio event loop after the process is forked, or otherwise is using as asyncio event loop that isn't friendly to forked processes.  In most cases after a fork, there shouldn't be an active asyncio event loop, which is why aiomultiprocess creates one when initializing the child process.  That said, it's not difficult to see if there's an existing event loop active first, I've created issue #5 to track that.
